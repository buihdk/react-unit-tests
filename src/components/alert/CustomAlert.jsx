import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { Alert } from 'antd';

const styles = {};

const CustomAlert = ({ className, message, description }) => (
  <Alert
    className={classnames(styles.alert, className)}
    message={message && <span className={styles.message}>{message}</span>}
    description={
      description && <span className={styles.description}>{description}</span>
    }
  />
);

CustomAlert.propTypes = {
  className: PropTypes.string,
  message: PropTypes.string,
  description: PropTypes.string,
};

CustomAlert.defaultProps = {
  className: '',
  message: '',
  description: '',
};

export default CustomAlert;
