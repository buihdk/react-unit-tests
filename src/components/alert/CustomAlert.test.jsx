import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CustomAlert from './CustomAlert';

describe(`CustomAlert`, () => {
  test(`renders without crashing`, () => {
    const wrapper = shallow(
      <CustomAlert message="Message" description="Description" />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
