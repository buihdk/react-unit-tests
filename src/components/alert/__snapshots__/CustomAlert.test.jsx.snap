// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert renders without crashing 1`] = `
<Alert
  className=""
  description={
    <span>
      Description
    </span>
  }
  message={
    <span>
      Message
    </span>
  }
/>
`;
